helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java

```diff
@@ -115,6 +113,11 @@ public boolean isCurrentOwner() {
         .getTimeout());
   }
 
+  @Override
+  public void close() {
+    _baseDataAccessor.close();
```

Comment:
You don't call unlock() here?

Reply:
Discussed offline. Won't unlock the lock inside close(), but will check to make sure the lock is unlocked when the user calls close().